src/Strategy/RPM.hs

```diff
@@ -0,0 +1,132 @@
+module Strategy.RPM (
+  discover,
+  getSpecDeps,
+  getTypeFromLine,
+
+  RPMDependency(..),
+  RequiresType(..),
+  Dependencies(..)
+) where
+
+import Prologue
+
+import Control.Effect.Error
+import qualified Data.Map.Strict as M
+import Data.Maybe (mapMaybe)
+import qualified Data.Text as T
+import DepTypes
+import Discovery.Walk
+import Effect.Grapher
+import Effect.ReadFS
+import Graphing (Graphing)
+import Types
+
+newtype SpecFileLabel =
+  RequiresType DepEnvironment
+  deriving (Eq, Ord, Show, Generic)
+  
+data RPMDependency = RPMDependency 
+  { rpmDepName :: Text,
+    rpmConstraint :: Maybe VerConstraint
+  } deriving (Eq, Ord, Show, Generic)
+
+{- These two structures should make it very easy to add other headers like `Requires`
+-}
+newtype RequiresType =
+    BuildRequires RPMDependency
+    deriving (Eq, Ord, Show, Generic)
+
+newtype Dependencies = Dependencies
+    { depBuildRequires :: [RPMDependency] 
+    } deriving (Eq, Ord, Show, Generic)
+{-
+Yep...  Those ones right above me.  Kinda rude.  They're actually standing on my head.
+-}
+
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \dir _ files ->
+  case find (\f -> ".spec" `isSuffixOf` fileName f) files of
+    Nothing -> pure WalkContinue
+    Just specFile -> do
+      runSimpleStrategy "rpm-spec" RustGroup $ fmap (mkProjectClosure dir) (analyze specFile)
+      pure WalkSkipAll
+
+analyze :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m) => Path Rel File -> m (Graphing Dependency)
+analyze specFile = do
+  specFileText <- readContentsText specFile
+  pure . buildGraph $ getSpecDeps specFileText
```

Comment:
In this context, `pure . buildGraph` is exactly equivalent to `pure $ buildGraph`.  However, `buildGraph $ getSpecDeps specFileText` requires a `$`, since the RHS of the `$` is type `a`, rather than `a -> b` (i.e.: it's a value, rather than a function).  I prefer to use `.` as much as possible, since it's much smaller and visually stays out of the way of the names which I care a lot more about.